AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,692 @@
+                self.messageLabel.alpha = 1.0
+            })
+        }
+
+        // Hide the message with a delay.
+        self.messageTimer?.invalidate()
+        let timer = Timer.init(timeInterval: TimeInterval(6), repeats: false) { (timer) in
+            if (self.messageLabel.isHidden == false) {
+                UIView.animate(withDuration: 0.6, animations: {
+                    self.messageLabel.alpha = 0
+                }, completion: { (complete) in
+                    if (complete) {
+                        self.messageLabel.isHidden = true
+                    }
+                })
+            }
+        }
+
+        self.messageTimer = timer
+        RunLoop.main.add(timer, forMode: .commonModes)
+    }
+
+    // MARK: Speech Recognition
+    func stopRecognizingAudio() {
+        if let recognizer = self.speechRecognizer {
+            recognizer.stopRecognizing()
+            self.speechRecognizer = nil
+
+            if let view = self.speechRecognizerView {
+                hideSpeechRecognitionUI(view: view)
+            }
+        }
+    }
+
+    func recognizeRemoteAudio(gestureRecognizer: UIGestureRecognizer) {
+        guard let remoteView = gestureRecognizer.view else {
+            print("Couldn't find a view attached to the tap recognizer. \(gestureRecognizer)")
+            return;
+        }
+        guard let room = self.room else {
+            print("We are no longer connected to the Room!")
+            return
+        }
+
+        // Find the Participant.
+        let hashedSid = remoteView.tag
+        for remoteParticipant in room.remoteParticipants {
+            for videoTrackPublication in remoteParticipant.remoteVideoTracks {
+                if (videoTrackPublication.trackSid.hashValue == hashedSid) {
+                    if let audioTrack = remoteParticipant.remoteAudioTracks.first?.remoteTrack {
+                        recognizeRemoteParticipantAudio(audioTrack: audioTrack,
+                                                        sid: remoteParticipant.remoteAudioTracks.first!.trackSid,
+                                                        name: remoteParticipant.identity,
+                                                        view: remoteView)
+                    }
+                }
+            }
+        }
+    }
+
+    func recognizeRemoteParticipantAudio(audioTrack: TVIRemoteAudioTrack, sid: String, name: String, view: UIView) {
+        if (self.speechRecognizer != nil) {
+            stopRecognizingAudio()
+        } else {
+            showSpeechRecognitionUI(view: view, message: "Listening to \(name)...")
+
+            recognizeAudio(audioTrack: audioTrack, identifier: sid)
+        }
+    }
+
+    func recognizeLocalAudio() {
+        if (self.speechRecognizer != nil) {
+            stopRecognizingAudio()
+        } else if let audioTrack = self.localAudioTrack {
+            // TODO: CE - Only allow this operation when we are in a Room, and recording audio.
```

Comment:
After discussing with @ptankTwilio we will update the comment to reference the known issue, but won't make any other changes.